lib/src/DynamicProto.ts

```diff
@@ -135,7 +135,7 @@ function _getObjProto(target:any) {
         }
 
         // target[Constructor] May break if the constructor has been changed or removed
-        let newProto = target[str__Proto] || target[Prototype] || target[Constructor];
+        let newProto = target[str__Proto] || target[Prototype] || (target[Constructor] ? target[Constructor][Prototype] : null);
```

Comment:
This is the root location of the issue, when IE is running in IE7 mode `target.__proto__` (target[str__Proto]) and `target.prototype` (target[Prototype]) don't exist on the root prototype and the final fallback of the target.constructor is also incomplete.

This issue ONLY affects IE7... Which is not strictly supported, but this was causing an endless loop in `_checkPrototype()`